### DISCUSSION TITLE
A Method to Collect Stats About Canceled Trips and their Resulting Delays

### DISCUSSION THREAD

**Discussion by ErikUden** [Traewelling/traewelling#2051] [Ideas / Feature Requests]
**Is your feature request related to a problem? Please describe.**
Since the route you take to get somewhere sometimes needs to be reconsidered due to cancelations of stops, it'd be nice if Träwelling could collect stats about the routes you _would've_ taken and how much time was lost due to the cancelations. 

When checking into a train long before it departures, Träwelling gives you a warning notifying you of the fact that you didn't get any points for checking in so early, but what if you want to plan your trip(s) ahead and or let Träwelling collect statistics about the time lost due to cancelations and changes in plan? 

**Describe the solution you'd like**
It'd be nice if one could plan their trip ahead within Träwelling, possibly “Add” trips before checking in, so that if a ride gets canceled Träwelling knows what other rides you needed to take to end at the same destination. 

I imagine it like this: 
Instead of only having a check-in button, you can also have the option to “Add to trip” so that you can create Trips and plan your travels ahead within Traveldroid/Träwelling. A trip is simply a collection of different rides. Your “trip” will be saved but any ride inside of it won't be counted as checked in. Träwelling now knows your whole plan, not just the individual ride from one train/bus station to the next.

Not only can Träwelling now calculate the waiting time on train / bus stations inbetween rides, but also can a _real_ delay time be measured. If a train gets canceled and you need to edit your trip because of it, then the time between your initial arrival (*without any cancelations*) and your edited arrival (*with the new rides you needed to take due to cancelations*) can be counted as delay time that was caused due to train cancelations. 

Also, when editing the trip and “deleting” a ride from it, you should be able to mention a reason such as “Couldn't catch it due to prior train being late” or “Was canceled” (*if that wasn't detected automatically, etc.*). 

With this Trip system, Check-ins could be easier than ever. Träwelling could even send push notifications saying “According to your trip, you should now be in the [Train] to [Place] do you want to check-in?”. Also, the top of the home screen can just display whatever the trip says should be checked-into next, so checking in could be as easy as opening the app and pressing a singular button instead of searching for your exact ride in the moment you have to take it! For those who plan in advance or have time to add all rides at the beginning of the trip, a great tool! 

As I do not know how the DB API works this is just a mere fantasy, however, trips could also be imported using the ticket system from the DB Navigator app!

When you want to import a ticket into the DB Navigator app, all you'd need is the six letter code and your surname / the surname of the passenger who bought that ticket. You can see your ticket code at the top left corner inside of the DB Navigator app when viewing your ticket. It would be nice if you could enter your ticket code and last name into Träwelling, which then uses the DB API to fetch that ticket's info and imports that as a ride. 

This way adding Trips wouldn't be like adding every singular ride of a trip that the DB Navigator app just searched out automagically, but easily imported. 

**Describe alternatives you've considered**
Allow check-ins many hours or even days before the ride starts but only display it once the ride starts, and if it is canceled let a user choose a replacement ride, this way similar statistics can be collected. 

**Additional context**
N/A

Thank you very much for reading! I love Träwelling ^ ^!!

**Comment by pascalwittler** [comment]
possible overlap with #2021
